src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/GetAllTest.java

```diff
@@ -35,34 +39,71 @@
+            .andExpect(contentIsArray)
+            .andExpect(contentIsEmpty)
+            .andExpect(pageIsLast)
+            .andExpect(pageIsFirst);
         sendGet("/jobs?page=1")
             .andExpect(status().isOk())
-            .andExpect(content().json(emptyResponse));
+            .andExpect(contentIsArray)
+            .andExpect(contentIsEmpty)
+            .andExpect(pageIsLast)
+            .andExpect(pageIsFirst);
         sendGet("/jobs?size=1")
             .andExpect(status().isOk())
-            .andExpect(content().json(emptyResponse));
+            .andExpect(contentIsArray)
+            .andExpect(contentIsEmpty)
+            .andExpect(pageIsLast)
+            .andExpect(pageIsFirst);
         sendGet("/jobs?page=1&size=1")
             .andExpect(status().isOk())
-            .andExpect(content().json(emptyResponse));
+            .andExpect(contentIsArray)
+            .andExpect(contentIsEmpty)
+            .andExpect(pageIsLast)
+            .andExpect(pageIsFirst);
     }
 
     @Test
-    public void should_return_some_data_when_non_empty_list_is_returned() throws Exception {
+    public void should_return_default_pages_when_no_parameters_are_passed() throws Exception {
         JobData jobData = JobData.fromJob("some-id", validJob());
-        when(jobsService.getAll(anyString())).thenReturn(new JobList(Collections.singletonList(jobData)));
+        when(jobsService.getAll(anyString(), anyInt(), anyInt()))
+            .thenReturn(new PageImpl<>(Collections.singletonList(jobData)));
 
         sendGet()
             .andExpect(status().isOk())
-            .andExpect(jsonPath("data[0].action").exists())
-            .andExpect(jsonPath("data[0].id").value(jobData.id))
-            .andExpect(jsonPath("data[0].name").value(jobData.name));
+            .andExpect(jsonPath("content[0].action").exists())
+            .andExpect(jsonPath("content[0].id").value(jobData.id))
+            .andExpect(jsonPath("content[0].name").value(jobData.name))
+            .andExpect(jsonPath("totalPages").value(1))
+            .andExpect(jsonPath("totalElements").value(1))
+            .andExpect(jsonPath("numberOfElements").value(1));
+    }
+
+    @Test
+    public void should_paginate_response_content_correctly_according_to_parameters() throws Exception {
+        getResponse(1, 10)
+            .andExpect(jsonPath("content").isEmpty())
+            .andExpect(jsonPath("totalPages").value(1))
+            .andExpect(jsonPath("numberOfElements").value(0));
+        getResponse(0, 2)
+            .andExpect(jsonPath("totalPages").value(2))
+            .andExpect(jsonPath("numberOfElements").value(2));
+        getResponse(1, 2)
+            .andExpect(jsonPath("totalPages").value(2))
+            .andExpect(jsonPath("numberOfElements").value(1));
+        getResponse(1, 1)
+            .andExpect(jsonPath("totalPages").value(3))
+            .andExpect(jsonPath("numberOfElements").value(1));
```

Comment:
I think it would be clearer if we'd set up how many jobs exist in 'given' step of this test

Reply:
thought about separating test cases. didn't appeal to me too much. but it's a no brainer to do so

Reply:
Can put a comment or so. It is a single test suite with singular purpose - can scroll down to check the total. But I'll put it in the test "title"